packages/react/src/components/Grid/Grid.js

```diff
@@ -5,58 +5,18 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import cx from 'classnames';
 import PropTypes from 'prop-types';
-import React, { useContext } from 'react';
+import React from 'react';
 import { useFeatureFlag } from '../FeatureFlags';
-import { usePrefix } from '../../internal/usePrefix';
-
-const SubgridContext = React.createContext(false);
-
-function Grid({
-  as: BaseComponent = 'div',
-  condensed = false,
-  narrow = false,
-  fullWidth = false,
-  columns = 16,
-  className: containerClassName,
-  children,
-  ...rest
-}) {
-  const prefix = usePrefix();
-  const hasCSSGrid = useFeatureFlag('enable-css-grid');
-  const isSubgrid = useContext(SubgridContext);
-
-  const cssGridClassNames = {
-    [`${prefix}--css-grid`]: !isSubgrid,
-    [`${prefix}--css-grid--${columns}`]: !isSubgrid && columns !== 16,
-    [`${prefix}--css-grid--condensed`]: condensed,
-    [`${prefix}--css-grid--narrow`]: narrow,
-    [`${prefix}--css-grid--full-width`]: fullWidth,
-    [`${prefix}--subgrid`]: isSubgrid,
-    [`${prefix}--col-span-${columns}`]:
-      (isSubgrid && columns !== 16) || columns !== 16,
-  };
-
-  const flexGridClassNames = {
-    [`${prefix}--grid`]: true,
-    [`${prefix}--grid--condensed`]: condensed,
-    [`${prefix}--grid--narrow`]: narrow,
-    [`${prefix}--grid--full-width`]: fullWidth,
-  };
-
-  const className = cx(
-    containerClassName,
-    hasCSSGrid ? cssGridClassNames : flexGridClassNames
-  );
-
-  return (
-    <SubgridContext.Provider value={true}>
-      <BaseComponent className={className} {...rest}>
-        {children}
-      </BaseComponent>
-    </SubgridContext.Provider>
-  );
+import { CSSGrid } from './CSSGrid';
+import { FlexGrid } from './FlexGrid';
+
+function Grid(props) {
+  const enableCSSGrid = useFeatureFlag('enable-css-grid');
+  if (enableCSSGrid) {
+    return <CSSGrid {...props} />;
+  }
+  return <FlexGrid {...props} />;
```

Comment:
I thought this wouldn't be behind a feature flag anymore and the two could coexist at the same time since they no longer share a name(space)?

I assume the goal is for FlexGrid to eventually be deprecated and removed, but from a usage standpoint this means that for all of v11 users will have to wrap every usage in a `<FeatureFlag>` toggle.

Reply:
@tay1orjones I think you're dead-on, I'm thinking the timeline is:

- In v10, we default to `FlexGrid` but will enable `CSSGrid` under the `Grid` namespace
- In v11, we can safely remove `FlexGrid` from `Grid` and rename `CSSGrid` to `Grid`. `FlexGrid` is also an export from the package

Reply:
@joshblack Ah, yes that makes more sense now. Sounds solid!